Code/Framework/AzFramework/Tests/DeviceAttributeSystemComponentTests.cpp

```diff
@@ -0,0 +1,158 @@
+        {
+            return m_description;
+        }
+
+        bool Evaluate(AZStd::string_view rule) const override
+        {
+            return m_eval(rule);
+        }
+
+        AZStd::any GetValue() const override
+        {
+            return m_value; 
+        }
+
+    private:
+        AZStd::string m_name;
+        AZStd::string m_description;
+        EvalFunc m_eval;
+        AZStd::any m_value;
+    };
+
+
+    TEST_F(DeviceAttributesSystemComponentTestFixture, DeviceAttributesSystem_Can_Register_Device_Attributes)
+    {
+        AZStd::string name { "TestAttribute" };
+        AZStd::string description { "Description" };
+        AZStd::string valueString{ "ABC123" };
+        AZStd::any value{ valueString };
+        TestDeviceAttribute::EvalFunc eval = [](AZStd::string_view rule)
+        {
+            // simple string comparison evaluator
+            return rule == "True";
+        };
+        auto deviceAttribute = AZStd::make_unique<TestDeviceAttribute>(name, description, value, eval);
+
+        // when the device attribute registrar exists
+        auto registrar = AzFramework::DeviceAttributeRegistrar::Get();
+        ASSERT_NE(registrar, nullptr);
+
+        // expect a device attribute can be registered
+        EXPECT_TRUE(registrar->RegisterDeviceAttribute(AZStd::move(deviceAttribute)));
+
+        // expect the registered attribute can be found
+        auto foundAttribute = registrar->FindDeviceAttribute(name);
+        EXPECT_NE(foundAttribute, nullptr);
+        auto foundValue = foundAttribute->GetValue();
+        EXPECT_EQ(foundValue.type(), azrtti_typeid<AZStd::string>());
+        EXPECT_EQ(AZStd::any_cast<AZStd::string>(foundValue), valueString);
+        EXPECT_EQ(foundAttribute->GetDescription(), description);
+        EXPECT_EQ(foundAttribute->GetName(), name);
+        EXPECT_TRUE(foundAttribute->Evaluate("True"));
+        EXPECT_FALSE(foundAttribute->Evaluate("Not true"));
+
+        // when another device attribute with the same attribute name is registered
+        auto deviceAttributeDuplicate = AZStd::make_unique<TestDeviceAttribute>(name, description, value, eval);
+
+        // expect failure because attribute names must be unique
+        EXPECT_FALSE(registrar->RegisterDeviceAttribute(AZStd::move(deviceAttributeDuplicate)));
+    }
+
+    TEST_F(DeviceAttributesSystemComponentTestFixture, DeviceAttributesSystem_Registers_Common_Device_Attributes)
+    {
+        // when the device attributes system component is activated
+
+        // expect a device attribute registrar exists
+        auto registrar = AzFramework::DeviceAttributeRegistrar::Get();
+        ASSERT_NE(registrar, nullptr);
+
+        // expect common attributes are registered
+        auto deviceModel = registrar->FindDeviceAttribute("DeviceModel");
+        EXPECT_NE(deviceModel, nullptr);
+        auto deviceModelValue = deviceModel->GetValue();
+        EXPECT_EQ(deviceModelValue.type(), azrtti_typeid<AZStd::string>());
```

Comment:
We should also check that the Device Model name is not an empty string.
```suggestion
        auto deviceModelValue = deviceModel->GetValue();
        ASSERT_TRUE(deviceModelValue.is<AZStd::string>());
        EXPECT_FALSE(AZStd::any_cast<AZStd::string>(deviceModelValue).empty());
```